spec/requests/api/v0/uni_can_pay_for_subscription_spec.rb

```diff
@@ -0,0 +1,53 @@
+RSpec.describe Api::V0::SubscriptionsController, type: :request do
+  describe 'POST /api/v0/subscriptions (Happy Path)' do
+    let(:university) { create(:user, role: :university) }
+    let(:credentials) { university.create_new_auth_token }
+    let(:headers) { { HTTP_ACCEPT: 'application/json' }.merge!(credentials) }
+    before do
+      StripeMock.start
+      post '/api/v0/subscriptions',
+           headers: headers,
+           params: { stripeToken: StripeMock.generate_card_token }
+    end
+    after { StripeMock.stop }
+
+  descirbe 'Payment posts successfully (Happy Path)' do
+    it 'return a 200 status' do
+      expect(response.status).to eq 200
+    end
+
+    it 'returns a success message' do
+      expect(response_json['message']).to eq 'Payment successful'
+    end
+  end
+
+  describe 'Payment does not post successfully (Sad Path)' do
```

Comment:
Okay, so these tests are in a way relevant, but they are NOT request specs, and should not be in this file. If you are interested I can elaborate on the subject in a session. If you want to test the sad path, you should make a post request without a `stripeToken` param.